AK/StringView.cpp

```diff
@@ -137,6 +138,18 @@ bool StringView::contains(char needle) const
     return false;
 }
 
+bool StringView::contains(u32 needle) const
+{
+    StringBuilder needle_builder;
```

Comment:
A method named `contains` performing an allocation would be very surprising to me.

Can we instead put an `AK::Array<char, 4>` on the stack here? Then use `AK::UnicodeUtils::code_point_to_utf8` to encode the code point into the array. You can then invoke the `contains(StringView)` overload on that array.

Something like (untested):
```c++
Array<char, 4> encoded;
size_t index = 0;

auto length = AK::UnicodeUtils::code_point_to_utf8(needle, [&](auto byte) {
    encoded[index++] = byte;
});

if (length == -1)
    return false;

return contains(StringView { encoded.data(), length });
```

Reply:
There is no allocation here. In fact, I implemented the local-char-array you describe completely, before I noticed that StringBuilder already has an inline-buffer.

Reply:
Should there be a comment explaining that then? As I think many folks would read this code and assume the same. Especially with a call to a `try_` function.

Reply:
Given that we now have `NoAllocationGuard`, how about we chuck that in there?

Reply:
NoAllocationGuard doesn't easily work, since for some reason it can't be used by the Kernel(?!) or DynamicLoader. So I added a comment, like you suggested.